### ISSUE TITLE
ModuleNotFoundError: No module named 'conda_build' with conda-build 3.21.3

### ISSUE THREAD

**Issue by parviste** [conda/conda-build#4182]
In our build pipeline we create and activate a conda environment using the following yaml:

```
name: build_env
channels:
  - defaults
dependencies:
  - conda-build=3.21.3
```

We then run

`conda index`

This results in the error:

```
Traceback (most recent call last):
  File "/home/jenkins/conda/envs/build_env/bin/conda-index", line 7, in <module>
    from conda_build.cli.main_index import main
ModuleNotFoundError: No module named 'conda_build'
```

If we pin the version of conda-build to `conda-build=3.21.1` in the environment yaml, no error is produced.

### Actual Behavior

An exception is raised when running `conda index`

### Expected Behavior

No exception is raised, conda_build module is found

### Steps to Reproduce

```sh
cat <<EOF >build_env.yaml
name: build_env
channels:
- defaults
dependencies:
- conda-build=3.21.3
EOF

conda env create -f build_env.yaml
conda activate build_env
conda index
```

##### `Output of conda info`

```

     active environment : build_env
    active env location : /home/parviste/miniconda3/envs/build_env
            shell level : 1
       user config file : /home/parviste/.condarc
 populated config files : /home/parviste/.condarc
          conda version : 4.8.3
    conda-build version : not installed
         python version : 3.8.3.final.0
       virtual packages : __glibc=2.31
       base environment : /home/parviste/miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /home/parviste/miniconda3/pkgs
                          /home/parviste/.conda/pkgs
       envs directories : /home/parviste/miniconda3/envs
                          /home/parviste/.conda/envs
               platform : linux-64
             user-agent : conda/4.8.3 requests/2.23.0 CPython/3.8.3 Linux/4.19.104-microsoft-standard ubuntu/20.04.1 glibc/2.31
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False

```

**Comment by mingwandroid** [comment]
Thanks for the bug report. I just cut a new build a few minutes before this report so it'll have to be addressed in the next build instead.

**Comment by mingwandroid** [comment]
Do you have a workaround at present too? Not that having one means I will ignore this issue, far from it :-)

**Comment by parviste** [comment]
The workaround is that we simply fix the version at 3.21.1 which seems to be the last working version. We have a bunch of repositories where we have not specified a version constraint, so it involves some work which we would of course prefer to avoid. But nothing critical.

**Comment by parviste** [comment]
It seems that conda-build=3.21.4 also works, so seems like it's only 3.21.3 that is somehow broken

**Comment by jezdez** [comment]
Closing since it was fixed in 3.21.4.
